src/api/action/search.ts

```diff
@@ -0,0 +1,122 @@
+import { isActor } from '@fedify/fedify';
+import { Context } from 'hono';
+import ky from 'ky';
+import sanitizeHtml from 'sanitize-html';
+
+import {
+    type HonoContextVariables,
+    fedify
+} from '../../app';
+
+interface ProfileSearchResult {
+    actor: any;
+    handle: string;
+    followerCount: number;
+    isFollowing: boolean;
+}
+
+interface SearchResults {
+    profiles: ProfileSearchResult[];
+}
+
+// @<username>@<domain>.<tld>
+const HANDLE_REGEX = /^@([\w-]+)@([\w-]+\.[\w.-]+)$/;
+
+// http(s)://...
+const URI_REGEX = /^https?:\/\/[^\s/$.?#].[^\s]*$/;
+
+export async function searchAction(
+    ctx: Context<{ Variables: HonoContextVariables }>,
+) {
+    const db = ctx.get('db');
+    const apCtx = fedify.createContext(ctx.req.raw as Request, {
+        db,
+        globaldb: ctx.get('globaldb'),
+    });
+
+    // Parse "query" from query parameters
+    const query = ctx.req.query('query') || '';
+
+    // Init search results - At the moment we only support searching for an actor (ui calls them profiles)
+    const results: SearchResults = {
+        profiles: [],
+    };
+
+    // If the query is not a handle or URI, return early
+    if (HANDLE_REGEX.test(query) === false && URI_REGEX.test(query) === false) {
+        console.log(`Invalid query: ${query}`);
+
+        return new Response(JSON.stringify(results), {
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            status: 200,
+        });
+    }
+
+    // Lookup actor by handle or url
+    try {
+        const actor = await apCtx.lookupObject(query);
```

Comment:
@allouis I was noticing that if `lookupObject` errors out (i.e a malformed url is passed), it doesn't get caught in this try / catch, which I would of expected?

Reply:
I guess it returns `null` instead of throwing? Or is there something else here?

Reply:
Yeah, it returns `null` instead of throwing!

Reply:
I was actually wanting to stop verbose output in the logs 😅 So at the moment if I passed this `@foo@bar.baz` I get 

```
at node:internal/deps/undici/undici:13178:13
     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
     at async lookupWebFinger (file:///opt/activitypub/node_modules/@fedify/fedify/esm/webfinger/lookup.js:32:24)
     at async lookupObject (file:///opt/activitypub/node_modules/@fedify/fedify/esm/vocab/lookup.js:58:21)
     at async searchAction (/opt/activitypub/src/api/action/search.ts:59:23)
     at async dispatch (file:///opt/activitypub/node_modules/hono/dist/compose.js:29:17)
     at async <anonymous> (/opt/activitypub/src/app.ts:279:5)
     at async dispatch (file:///opt/activitypub/node_modules/hono/dist/compose.js:29:17)
     at async <anonymous> (/opt/activitypub/src/app.ts:259:5)
     at async dispatch (file:///opt/activitypub/node_modules/hono/dist/compose.js:29:17) {
   [stack]: 'TypeError: fetch failed\n' +
     '    at node:internal/deps/undici/undici:13178:13\n' +
     '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
     '    at async lookupWebFinger (file:///opt/activitypub/node_modules/@fedify/fedify/esm/webfinger/lookup.js:32:24)\n' +
     '    at async lookupObject (file:///opt/activitypub/node_modules/@fedify/fedify/esm/vocab/lookup.js:58:21)\n' +
     '    at async searchAction (/opt/activitypub/src/api/action/search.ts:59:23)\n' +
     '    at async dispatch (file:///opt/activitypub/node_modules/hono/dist/compose.js:29:17)\n' +
     '    at async <anonymous> (/opt/activitypub/src/app.ts:279:5)\n' +
     '    at async dispatch (file:///opt/activitypub/node_modules/hono/dist/compose.js:29:17)\n' +
     '    at async <anonymous> (/opt/activitypub/src/app.ts:259:5)\n' +
     '    at async dispatch (file:///opt/activitypub/node_modules/hono/dist/compose.js:29:17)',
   [message]: 'fetch failed',
   [cause]: Error: getaddrinfo ENOTFOUND bar.baz
       at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26) {
     [stack]: 'Error: getaddrinfo ENOTFOUND bar.baz\n' +
       '    at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:120:26)',
     [message]: 'getaddrinfo ENOTFOUND bar.baz',
     errno: -3008,
     code: 'ENOTFOUND',
     syscall: 'getaddrinfo',
     hostname: 'bar.baz'
   }
 }
 ```

Reply:
Ah, its because Fedify logs it as part of `DBG` - ` 06:47:26.790 DBG fedify·webfinger·lookup Failed to fetch WebFinger resource descriptor: TypeError: fetch failed`

Ignore me 😄